lib/Doctrine/DBAL/Connection.php

```diff
@@ -1499,7 +1501,7 @@ private function getBindingInfo($value, $type)
             $value       = $type->convertToDatabaseValue($value, $this->getDatabasePlatform());
             $bindingType = $type->getBindingType();
         } else {
-            $bindingType = $type;
+            $bindingType = $type ?? ParameterType::NULL;
```

Comment:
Should we just not specify any type in this case? The less DBAL implies, the better.

Reply:
Probably, but it's out of scope and will require some refactoring (results of this method are used as list() now).

Reply:
As far as I know, a common default for a binding type is "string", not "null". Binding as "null" by default may have side effects.

It's a private method, so there shouldn't be much refactoring. In general, while working on the latest refactoring issues, I found most of the methods returning a list really smelling. Why instead of returning some unstructured set of values it doesn't just _do_ the binding?